prow/plugins/updateconfig/updateconfig.go

```diff
@@ -123,26 +133,46 @@ func handle(gc githubClient, kc kubeClient, log *logrus.Entry, pre github.PullRe
 		return err
 	}
 
-	message := func(cm plugins.ConfigMapSpec, fileName string) string {
-		msg := fmt.Sprintf("`%s` configmap from file `%s`", cm.Name, fileName)
-		if cm.Namespace != "" {
-			msg = fmt.Sprintf("%s in namespace `%s`", msg, cm.Namespace)
+	message := func(name, namespace string, data map[string]string) string {
+		identifier := fmt.Sprintf("`%s` configmap", name)
+		if namespace != "" {
+			identifier = fmt.Sprintf("%s in namespace `%s`", identifier, namespace)
+		}
+		msg := fmt.Sprintf("%s using the following files:", identifier)
+		for key, file := range data {
+			msg = fmt.Sprintf("%s\n - key `%s` using file `%s`", msg, key, file)
 		}
 		return msg
 	}
 
 	// Are any of the changes files ones that define a configmap we want to update?
 	var updated []string
+	type configMapID struct {
+		name, namespace string
+	}
+	toUpdate := map[configMapID]map[string]string{}
 	for _, change := range changes {
 		cm, ok := configMaps[change.Filename]
 		if !ok {
 			continue // This file does not define a configmap
 		}
 		// Yes, update the configmap with the contents of this file
-		if err := update(gc, kc, org, repo, *pr.MergeSHA, change.Filename, cm.Name, cm.Namespace); err != nil {
+		key := cm.Key
+		if key == "" {
+			key = path.Base(change.Filename)
+		}
+		id := configMapID{name: cm.Name, namespace: cm.Namespace}
+		if _, ok := toUpdate[id]; !ok {
+			toUpdate[id] = map[string]string{}
+		}
+		toUpdate[id][key] = change.Filename
```

Comment:
Hummm, I think if you have multiple keys in the configmap, and only have one key has changed, the other unchanged keys will be removed by calling replace configmap. Might need to change it to patch? Or we need to fetch each config file regardless..

Reply:
That's why I was selectively only updating the keys that were changed but using a `GET` first to load the entire data?

Reply:
ah sorry missed that part :-)

With this I can probably just use a regex match to update all affected files without list the entire dir, and just need to handle deletions